dspace-api/src/main/java/org/dspace/external/provider/impl/SHERPAv2JournalDataProvider.java

```diff
@@ -0,0 +1,358 @@
+        // httpclient 4.3+ doesn't appear to have any sensible defaults any more. Setting conservative defaults as
+        // not to hammer the SHERPA service too much.
+        client = builder
+            .disableAutomaticRetries()
+            .setMaxConnTotal(5)
+            .build();
+    }
+
+    /**
+     * Get a single journal based on a "title equals string" query
+     * @param id    The journal title which will be used as query string
+     * @return external data object representing journal
+     */
+    @Override
+    public Optional<ExternalDataObject> getExternalDataObject(String id) {
+
+        HttpGet method = null;
+        SHERPAResponse sherpaResponse = null;
+        int timeout = 5000;
+        URIBuilder uriBuilder = null;
+        try {
+            // Construct URI for an exact match on journal title
+            uriBuilder = new URIBuilder(url);
+            uriBuilder.addParameter("item-type", "publication");
+            uriBuilder.addParameter("filter", "[[\"title\",\"equals\",\"" + id + "\"]]");
+            uriBuilder.addParameter("format", "Json");
+            if (StringUtils.isNotBlank(apiKey)) {
+                uriBuilder.addParameter("api-key", apiKey);
+            }
+
+            // Build HTTP method
+            method = new HttpGet(uriBuilder.build());
+            method.setConfig(RequestConfig.custom()
+                                          .setConnectionRequestTimeout(timeout)
+                                          .setConnectTimeout(timeout)
+                                          .setSocketTimeout(timeout)
+                                          .build());
+
+            // Execute the method
+            HttpResponse response = client.execute(method);
+            int statusCode = response.getStatusLine().getStatusCode();
+
+            if (statusCode != HttpStatus.SC_OK) {
+                sherpaResponse = new SHERPAResponse("SHERPA/RoMEO return not OK status: "
+                                                        + statusCode);
+            }
+
+            HttpEntity responseBody = response.getEntity();
+
+            // Get InputStream from API response and parse JSON
+            if (null != responseBody) {
+                InputStream content = null;
+                try {
+                    content = responseBody.getContent();
+                    sherpaResponse = new SHERPAResponse(content, SHERPAResponse.SHERPAFormat.JSON);
+                } catch (IOException e) {
+                    log.error("Encountered exception while contacting SHERPA/RoMEO: " + e.getMessage(), e);
+                } finally {
+                    if (content != null) {
+                        content.close();
+                    }
+                }
+            } else {
+                sherpaResponse = new SHERPAResponse("SHERPA/RoMEO returned no response");
+            }
+        } catch (Exception e) {
+            log.error("SHERPA/RoMEO query failed: ", e);
```

Comment:
I don't think we can.
This class and its methods implement ExternalDataService interface (via extending ExternalDataServiceImpl). The abstract methods don't have any throw statements in its signatures. I would be hesitant to include a refactor of this interface and abstract impl in this PR as it would have consequences for all other providers.